src/templates/console.html

```diff
@@ -19,74 +19,79 @@
     <script>
       languagePluginLoader.then(() => {
         pyodide.runPython(`
-            from pyodide import console
+            import sys
+            import builtins
             import js
+            from reprlib import Repr
+            from pyodide.console import InteractiveConsole
 
 
-            class Console(console.InteractiveConsole):
+            # limiting output size to 1000 to avoid console hang
+            _repr = Repr()
+            _repr.maxdict = _repr.maxlist = _repr.maxtuple = _repr.maxset = _repr.maxfrozenset = _repr.maxdeque = _repr.maxarray = _repr.maxlong = _repr.maxstring = _repr.maxother = 10 ** 3
+
+            def displayhook(value):
+                # from https://docs.python.org/3/library/sys.html#sys.displayhook
+                if value is None:
+                    return
+                builtins._ = None
+                text = _repr.repr(value)
+                try:
+                    sys.stdout.write(text)
+                except UnicodeEncodeError:
+                    bytes = text.encode(sys.stdout.encoding, 'backslashreplace')
+                    if hasattr(sys.stdout, 'buffer'):
+                        sys.stdout.buffer.write(bytes)
+                    else:
+                        text = bytes.decode(sys.stdout.encoding, 'strict')
+                        sys.stdout.write(text)
+                sys.stdout.write("\\n")
+                builtins._ = value
+
+            sys.displayhook = displayhook
+
+
+            class PyConsole(InteractiveConsole):
+                def __init__(self):
```

Comment:
I think we could still put this files in `pyodide-py/pyodide/console.py` to benefit from code formatting, etc. Just making them private with a leading `_` and indicate that they are used in this file should be enough?

Reply:
Only disadvantage of that is that increases the pyodide download size even for people who don't use the code. Probably not by enough to worry about though.

Reply:
Well, right now the minimal download size for pyodide is 6.5MB, compressed, this code snippet is 0.8kB compressed. So I think there lots of other things to optimize before we need to worry about it..

Reply:
Absolutely, I realize that was a bit of a silly remark.